packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,231 @@
+import debounce from 'debounce-promise';
+import differenceWith from 'lodash.differencewith';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import { useIntl } from 'react-intl';
+import { DragDropContext } from 'react-beautiful-dnd';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import CollapsibleMotion from '@commercetools-uikit/collapsible-motion';
+import FieldLabel from '@commercetools-uikit/field-label';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import {
+  CloseIcon,
+  EyeCrossedIcon,
+  EyeIcon,
+  SearchIcon,
+} from '@commercetools-uikit/icons';
+import AccessibleButton from '@commercetools-uikit/accessible-button';
+import { filterDataAttributes } from '@commercetools-uikit/utils';
+import DroppablePanel from '../droppable-panel';
+import {
+  Container,
+  HeaderContainer,
+  DroppableContainer,
+} from '../data-table-settings-panel.styles';
+import messages from './messages';
+
+export const ColumnManager = (props) => {
+  const { formatMessage } = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+  const handleDragEnd = (result) => {
+    setIsDragging(false);
+
+    // Invalid drop destination, do nothing
+    if (!result.destination) return;
+    if (result.destination.droppableId === 'hidden-columns-panel') {
+      if (result.source.droppableId === 'hidden-columns-panel') return;
+      props.onUpdateColumns([
+        ...props.selectedColumns.slice(0, result.source.index),
+        ...props.selectedColumns.slice(result.source.index + 1),
+      ]);
+    } else {
+      // the destination is the selected columns panel
+
+      // it's a swap when the source and the destination are the same
+      const isSwap = result.source.droppableId === 'selected-columns-panel';
+
+      const items = isSwap
+        ? // remove the dragged item from its position if it is not coming from
+          // the hidden section (it is a swap)
+          [
+            ...props.selectedColumns.slice(0, result.source.index),
+            ...props.selectedColumns.slice(result.source.index + 1),
+          ]
+        : props.selectedColumns;
+
+      const columns = isSwap ? props.selectedColumns : props.availableColumns;
+      const draggedColumn = columns.find(
+        (col) => col.key === result.draggableId
+      );
+
+      // push the column in the new position
+      props.onUpdateColumns([
```

Comment:
Should we check that draggedColumn is defined before trying to operate with it?  There's a chance that the result of `columns.find()` above could return `undefined`.

Reply:
We can add doble checking, but take into account that the only elements that can actually trigger a drop event are the column tags.